pkg/pointerdb/audit/audit_segment.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"context"
+	"crypto/rand"
+	"math/big"
+	"sync"
+
+	"github.com/vivint/infectious"
+
+	"storj.io/storj/pkg/eestream"
+	"storj.io/storj/pkg/paths"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/pointerdb/pdbclient"
+	"storj.io/storj/pkg/storage/meta"
+)
+
+// Audit to audit segments
+type Audit struct {
+	pointers pdbclient.Client
+	lastPath *paths.Path
+	mutex    sync.Mutex
+}
+
+// NewAudit creates a new instance of audit
+func NewAudit(pointers pdbclient.Client) *Audit {
+	return &Audit{
+		pointers: pointers,
+	}
+}
+
+// Stripe is a struct that contains stripe info
+type Stripe struct {
+	stripe int
+}
+
+// NextStripe returns a random stripe to be audited
+func (a *Audit) NextStripe(ctx context.Context) (stripe *Stripe, more bool, err error) {
+	a.mutex.Lock()
+	defer a.mutex.Unlock()
+
+	// retrieve a random list of pointers
+	var pointerItems []pdbclient.ListItem
+	var path paths.Path
+
+	// need to get random limit
+	if a.lastPath == nil {
+		pointerItems, more, err = a.pointers.List(ctx, nil, nil, nil, true, 0, meta.None)
+	} else {
+		pointerItems, more, err = a.pointers.List(ctx, nil, *a.lastPath, nil, true, 0, meta.None)
+	}
+
+	if err != nil {
+		return nil, more, err
+	}
+
+	randomNum, err := rand.Int(rand.Reader, big.NewInt(int64(len(pointerItems))))
+	if err != nil {
+		return nil, more, err
+	}
+	randomNumInt64 := randomNum.Int64()
+	pointerItem := pointerItems[randomNumInt64]
```

Comment:
can we extract the above lines to a helper function `getRandomPointer()`?

Reply:
I actually suggested the opposite, but with inline comments about the process.

Reply:
Although maybe yeah, everything upwards from there, could be separate.